stdlib/list.ysh

```diff
@@ -0,0 +1,55 @@
+func any(list) {
+  ## Returns true if any value in the list is truthy.
+  ##
+  ## If the list is empty, return false.
+
+  for item in (list) {
+    if (item) {
+      return (true)
+    }
+  }
+  return (false)
+}
+
+func all(list) {
+  ## Returns true if all values in the list are truthy.
+  ##
+  ## If the list is empty, return false.
```

Comment:
This should be true!  I think it falls out of boolean algebra, and intuitively you can say "all elements in an empty list are truthy" -- it's vacously true

Funny thing is that I think this came up on Hacker News recently - https://news.ycombinator.com/item?id=37264149

Yeah I agree with the top answer, it's algebra -- 

- 0 is the identity element for addition -- (x + 0 == x) for all x
- 1 is the identity element for multiplication -- (x * 1 == x) for all x
- false is the identity element for AND  -- (x AND false == x) for all x
  - all() is a reduce() with AND
- true is the identity element for OR -- (x OR true == x) for all x
  - any() is a reduce() with OR

Although I guess it's backward.  alltrue([]) == true and anytrue([]) == false



```
>>> any([])
False
>>> all([])
True
>>> 
```